app/service/planning_svc.py

```diff
@@ -47,9 +43,10 @@ def __init__(self):
         """
         link_status = await self._default_link_status(operation)
         links = []
-        if agent and await self._check_untrusted_agents_allowed(agent=agent, operation=operation,
+        if agent:
+            if await self._check_untrusted_agents_allowed(agent=agent, operation=operation,
                                                                 msg='no cleanup-link created'):
-            links.extend(await self._generate_cleanup_links(operation=operation, agent=agent, link_status=link_status))
+                links.extend(await self._generate_cleanup_links(operation=operation, agent=agent, link_status=link_status))
```

Comment:
This block and the block below in the else are identical blocks of code - should be a private function

Reply:
```
if await self._check_untrusted_agents_allowed(agent=agent, operation=operation,
                                                                msg='no cleanup-link created'):
                links.extend(
                    await self._generate_cleanup_links(operation=operation, agent=agent, link_status=link_status)
                    )
```

Reply:
yea I was on the fence on doing a DRY fix here, because its a minimal/clear repetition. Will change